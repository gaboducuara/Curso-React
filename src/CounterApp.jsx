import { useState } from 'react';
import PropTypes from "prop-types";
import { render } from 'react-dom';

export const CounterApp = ({ value }) => {
  
  const [ counter , setCounter] = useState(value);
  
  console.log(render)

  const handleAdd = ()  => {
  // console .log(newEvent);
    // setCounter (counter + 1)
    setCounter((c) => c + 1)
}

  const handleAddmenos = () => setCounter(counter - 1)
  const handleReset = () => setCounter( value ) 
        
    return (
      <>
        <h1> CounterApp </h1>
        <h3> { counter } </h3>

        <button onClick={ handleAdd }> +1 </button>
        <button onClick={handleAddmenos}> -1 </button>
        <button onClick={ handleReset }> Reset </button>
      </>
    );
  };

// ProtoTypes
CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};

//Mensaje por defecto si no se cumple el ProtoTypes
CounterApp.defaultProps = {
  value: "No se encuentra el numero",
};
