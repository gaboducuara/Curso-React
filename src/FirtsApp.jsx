import PropTypes  from 'prop-types'

export const FirtsApp = ( {
  title, 
  subTitle,
  name,
  lastName
} ) => {

  return (
    <>  
        <h1 data-testid="test-title">{title}</h1>
        <h2>{subTitle}</h2>
        <h2>{subTitle}</h2>
        <h3>{name}</h3>
        <h3>{lastName}</h3>
    </>  
  )
}
// LOS PROPTYPES
FirtsApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle:PropTypes.string.isRequired,
  name:PropTypes.string.isRequired,
  lastName:PropTypes.string.isRequired,
}

FirtsApp.defaultProps = {
  // title: 'NO HAY TITULO',
  subTitle:'No hay Subtitulo',
  name:'Gabriel',
  lastName:'Mancilla Ducuara'
}