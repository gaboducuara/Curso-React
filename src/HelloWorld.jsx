//     //COMPONENTE    
//  function App () {
//         // document.createElement
//         return (<h1>hola mundo!!! </h1>)
//     }

// export default App

export const HelloWorld = () => {
    return (
      <h1>hello world App</h1>
    )
  }