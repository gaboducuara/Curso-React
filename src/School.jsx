import PropTypes from 'prop-types'

export const School = ({name}) => {
  return (
    <>
        <h1>{name}</h1>
    </>
  )
}

School.propTypes = {
    name:PropTypes.string.isRequired
}

School.defaultProps = {
    name: 'El colegio no tiene nombre'
}