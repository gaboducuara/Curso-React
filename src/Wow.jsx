import PropTypes from 'prop-types'

export const Wow = ({ igor }) => {
  return (
    <>
        <h1>{igor}</h1>
    </>
  )
}

Wow.propTypes = {
    igor: PropTypes.string.isRequired
}

Wow.defaultProps = {
    igor:'no se encontro el nombre de bani'
}