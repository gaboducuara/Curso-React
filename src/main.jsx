//IMPORTACION REACT
import React from 'react'
//HERRAMIENTA REACTDOM PARA RENDERIZAR REACT
import ReactDOM from 'react-dom/client'

//COMPONENTES HA RENDERIZAR
// import { FirtsApp } from './FirtsApp'
   import { CounterApp } from './CounterApp'
// import { Wow } from './wow'
// import { School } from './School'

//IMPORTACION DE ESTILO GLOBAL EN LA APLICACION
import './styles.css'

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    {/* <FirtsApp /> */}
    <CounterApp value = {1}/>
    {/* <Wow igor = 'Mi nombre es bani'/> */}
    {/* <School name = 'instituto tecnico industrial'/> */}
  </React.StrictMode>,
)
