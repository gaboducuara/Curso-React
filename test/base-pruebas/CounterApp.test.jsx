import { fireEvent, getAllByText, render, screen } from '@testing-library/react';
import { CounterApp } from '../../src/CounterApp';

describe('Pruebas en <CounterApp />', () => {
    
    const initialvalue = 100;
    
    test('debe de hacer match con el snapshot', () => {

        //renderiza un componente en memoria
        const {container} = render( <CounterApp value={ initialvalue } /> )
        expect(container).toMatchSnapshot();
    });

    test('debe de mostrar el valor inicial de 100 <CounterApp value={100}', () => {
        render( <CounterApp value={100} /> )
        expect(screen.getByText(100)).toBeTruthy();
        // expect(screen.getByRole('headig', {level:2}).innerHTML).toContain('100')
    });

    test('Debe de incrementar con el boton +1', () => {

        render( <CounterApp value={ initialvalue } />);
        //fireEvent es un metodo para disparar eventos
        fireEvent.click(screen.getByText('+1'))
        //Seleccion respectiva
        expect(screen.getByText('101')).toBeTruthy();
    });

    test('Debe de Decrementar con el boton de -1', () => {

        render( <CounterApp value={ initialvalue } />);
        //fireEvent es un metodo para disparar eventos
        fireEvent.click(screen.getByText('-1'))
        //Seleccion respectiva
        expect(screen.getByText('99')).toBeTruthy();
    });

    
    test('Debe de funcionar con el boton de rest', () => {

        render( <CounterApp value={ initialvalue } />);
        fireEvent.click(screen.getByText('+1'))
        fireEvent.click(screen.getByText('+1'))
        fireEvent.click(screen.getByText('+1'))
        // fireEvent.click(screen.getByText('Reset'))
        fireEvent.click(screen.getByRole('button', {name:'btn-reset'}))
        expect(screen.getByText(initialvalue)).toBeTruthy();
    });
});