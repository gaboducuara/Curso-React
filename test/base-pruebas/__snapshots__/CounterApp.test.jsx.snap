// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp /> debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
     CounterApp 
  </h1>
  <h3>
     
    100
     
  </h3>
  <button>
     +1 
  </button>
  <button>
     -1 
  </button>
  <button>
     Reset 
  </button>
  <button
    aria-label="btn-reset"
  >
    Reset
  </button>
</div>
`;
